components/nexusoperations/executors.go

```diff
@@ -228,8 +229,9 @@ func (e activeExecutor) executeInvocationTask(ctx context.Context, env hsm.Envir
 			err := rawResult.Successful.Consume(&payload)
 			if err != nil {
 				callErr = err
+			} else if payload.Size() > e.Config.PayloadSizeLimit(ns.Name().String()) {
+				callErr = ErrResponseBodyTooLarge
```

Comment:
Make sure you test this and ensure that the operation ends up in a terminal failed state.